mempool/src/core_mempool/transaction_store.rs

```diff
@@ -263,6 +263,7 @@ impl TransactionStore {
                     // If the transaction is the same, it's an idempotent call
                     // Updating signers is not supported, the previous submission must fail
                     counters::CORE_MEMPOOL_IDEMPOTENT_TXNS.inc();
+                    self.process_ready_transactions(&address, acc_seq_num);
```

Comment:
if the 2nd time you receive the same txn but the acc seq number is up to date, you have to reprocess its readiness, otherwise it will be parked forever.

Reply:
note for future: we can probably clean things based on potentially new seq num.
```
        self.clean_committed_transactions(&address, acc_seq_num);
```